src/core/src/op/constant.cpp

```diff
@@ -542,6 +542,11 @@ void Constant::update_identical_flags(bool is_checked, bool identical_value) con
 
 void Constant::validate_and_infer_types() {
     set_output_type(0, m_element_type, m_shape);
+    if (const auto tensor = get_tensor_view()) {
```

Comment:
Why do we make an exception for the Constant node and put lower/upper propagation into the `validate_and_infer_types`? Can we keep the efficiency and put this code into the `evaluate_lower/upper`? Currently `evaluate_lower/upper` is unchanged -- why?

Reply:
Valid and infer set view on constant data, there is no data copy. Create view should not be cheap.
Constant data should never change after creation the bounds should be always available.

The evaluate_lower/upper make data copy to user tensors (is safe). The additional check  can be done as separate task if copy views or shared tensor is possible without any regression and avoid data copy.